doc/dev/adr/0023-large-message-chunking.md

```diff
@@ -0,0 +1,120 @@
+﻿# ADR 23: Large Message Chunking in MQTT Protocol
+
+## Status
+
+Proposed
+
+## Context
+
+The MQTT protocol has inherent message size limitations imposed by brokers and network constraints. Azure IoT Operations scenarios often require transmitting payloads that exceed these limits (e.g., firmware updates, large telemetry batches, complex configurations). Without a standardized chunking mechanism, applications must implement their own fragmentation strategies, leading to inconsistent implementations and interoperability issues.
+
+## Decision
+
+We will implement sdk-level message chunking as part of the Protocol layer to transparently handle messages exceeding the MQTT broker's maximum packet size.
+
+**The chunking mechanism will**:
+
+- Be enabled/disabled by a configuration setting.
+- Use standardized user properties for chunk metadata. The `__chunk` user property will contain a colon-separated string with chunking metadata: ```<messageId>:<chunkIndex>:<totalChunks>:<checksum>```. The string will include:
```

Comment:
* Speaking of versions - should this include a version in it of the chunking version?
  * How does mRPC command/response handle versions of the mRPC protocol layer itself (which is different than the customer mRPC client/servers having the version field)?  Can we leverage those concepts?
  * As one case I'm imagining, we could imagine(!) a v1 of this protocol requires sender to know size up front but then we relax in a v2 to support streaming.